microprofile/WFLY-13023_upgrade_microprofile-config_1.4.adoc

```diff
@@ -0,0 +1,58 @@
+= Upgrade MicroProfile Config to 1.4
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          microprofile,config,observability
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+
+MicroProfile 3.3 provides a minor upgrade for https://github.com/eclipse/microprofile-config/releases/tag/1.4[MicroProfile Config 1.4].
+
+This component upgrade also covers the upgrade of smallrye-config that implements MicroProfile Config 1.4.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-13023[WFLY-13023]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1427[EAP7-1427]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:mkopecky@redhat.com[Marek Kopecky]
+
+=== Testing By
+
+[X] QE
+
+== Requirements
+
+* Upgrade MicroProfile Config to 1.4
+* Upgrade smallrye-config to the release that implements MicroProfile Config 1.4
+* Run any MicroProfile Health test in WildFly integration test suite
+* Run the Eclipse MicroProfile Health TCK without failures
+
+=== Non-Requirements
+
+== Implementation Plan
+
+== Test Plan
+
+* Run the WildFly integration basic test suite and checks there are no failures related to MicroProfle Config tests.
+* Run the `microprofile-tck/config` module from WildFly integration test suite and checks that there are no failures.
+* Some tests that relates to the https://github.com/eclipse/microprofile-config/issues/446[inconsistent behaviour for empty/missing config property (#446)] are excluded as smallrye-config implementation does not pass the TCK tests that are challenged by this issue.
```

Comment:
> smallrye-config implementation does not pass the TCK tests

@jmesnil Can you please share more details about this? Which tests doesn't pass? I don't see any ignored tests in WF TS

> Some tests that relates to the https://github.com/eclipse/microprofile-config/issues/446[inconsistent behaviour for empty/missing config property (#446)] are excluded ...

Can you please share list of these tests?

> https://github.com/eclipse/microprofile-config/issues/446

I'm looking to this issue and it seems that there is no conclusion about future behaviour.

Reply:
@marekkopecky, test excluded from the MP config TCK are configured in https://github.com/wildfly/wildfly/blob/3af5d5438457bc2697ffc59118e7d5ce247f875d/testsuite/integration/microprofile-tck/config/tck-suite.xml#L11

The discussion in https://github.com/eclipse/microprofile-config/issues/446 is ongoing will not be resolved before MP Config 2.0 is released

Reply:
I check WF TS and SmallRye TS (1.6.1). Excluded TCK tests fails in both these TSs